src/api/spotify.ts

```diff
@@ -13,3 +15,28 @@ export function getUserData() {
 		return data;
 	};
 }
+
+export function getSongs(params: string) {
+	const xhr = new XMLHttpRequest();
+	xhr.open('GET', `https://api.spotify.com/v1/recommendations?${params}`, true);
+	xhr.setRequestHeader('Content-Type', 'application/json');
+	xhr.setRequestHeader('Authorization', `Bearer ${localStorage.getItem('access_token')}`);
+	xhr.send(null);
+	xhr.onload = () => {
+		const responseData = JSON.parse(xhr.responseText);
+		const songArray = responseData.items;
+		const songList: SongDetails[] = [];
+		const artists: string[] = responseData.artists.map((artist: any) => artist.name);
+		songArray.array.forEach((song: any) => {
```

Comment:
Why is this type `any`?

Reply:
I was going to ask if I should create interfaces for the expected structure. The song would look like the following:
interface Song {
  album_type: string;
  total_tracks: number;
  available_markets: string[];
  external_urls: {
    spotify: string;
  };
  href: string;
  id: string;
  images: {
    url: string;
    height: number;
    width: number;
  }[];
  name: string;
  release_date: string;
  release_date_precision: string;
  type: string;
  uri: string;
  artists: {
    external_urls: {
      spotify: string;
    };
    href: string;
    id: string;
    name: string;
    type: string;
    uri: string;
  }[];
}

Reply:
Yeah, you should go ahead and do that. Does spotify not have a package for this?